src/tatm/tokenizer/engine.py

```diff
@@ -117,6 +117,15 @@ def shutdown(self):
         self.shutdown_flag = True
         self.done = True
 
+    def list_source_datasets(self):
+        out = []
+        for dataset in self.datasets:
+            if isinstance(dataset, TatmTextData) and dataset.corpus:
+                out.append(f"{dataset.metadata.name}:{dataset.corpus}")
```

Comment:
Is it possible to use inheritance to remove the condition? Like add a function `get_source` to all Tatm Data instances?

Reply:
That's a good suggestion, let me see if I can get it in there